packages/next/server/web/utils.ts

```diff
@@ -1,17 +1,29 @@
 import type { NodeHeaders } from './types'
 
-export async function* streamToIterator<T>(
+export function readableStreamTee<T = any>(
   readable: ReadableStream<T>
-): AsyncIterableIterator<T> {
+): [ReadableStream<T>, ReadableStream<T>] {
+  const transformStream = new TransformStream()
+  const transformStream2 = new TransformStream()
+  const writer = transformStream.writable.getWriter()
+  const writer2 = transformStream2.writable.getWriter()
+
   const reader = readable.getReader()
-  while (true) {
-    const { value, done } = await reader.read()
-    if (done) break
-    if (value) {
-      yield value
-    }
+  function read() {
+    reader.read().then(({ done, value }) => {
+      if (done) {
+        writer.close()
+        writer2.close()
+        return
+      }
+      writer.write(value)
+      writer2.write(value)
+      read()
+    })
   }
-  reader.releaseLock()
+  read()
+
+  return [transformStream.readable, transformStream2.readable]
 }
```

Comment:
**question:** why is `readableStreamTee()` declared here? it's already exported by `node-web-stream-helper.ts` file

Reply:
Good catch! Removed duplicated code 🙂